### PR TITLE
Plfm 3600

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java [hunk 1]
```diff
@@ -893,41 +946,31 @@ public EntityHeader getEntityHeader(String nodeId, Long versionNumber) throws Da
...
+					throws SQLException {
+				EntityHeader header = new EntityHeader();
+				Long entityId = rs.getLong(COL_NODE_ID);
+				header.setId(KeyFactory.keyToString(entityId));
+				EntityType type = EntityType.valueOf((String) rs.getString(COL_NODE_TYPE));
+				header.setType(EntityTypeUtils.getEntityTypeClassName(type));
+				header.setName(rs.getString(COL_NODE_NAME));
+				header.setVersionNumber(rs.getLong(COL_REVISION_NUMBER));
+				header.setVersionLabel(rs.getString(COL_REVISION_LABEL));
+				header.setBenefactorId(rs.getLong(COL_NODE_BENEFACTOR_ID));
```
Comment (brucehoff) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java:964]: replace version label with version number to be consistent with the earlier query, please.

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/DBOAccessControlListDAOImplTest.java [hunk 2]
```diff
@@ -194,6 +197,51 @@ public void testCanAccess() throws Exception {
...
+	}
+	
+	@Test (expected=IllegalArgumentException.class)
+	public void testCanAccessMultipleBenefactorsNull() throws Exception {
+		Set<Long> gs = new HashSet<Long>();
+		gs.add(Long.parseLong(group.getId()));
+		Set<Long> benefactors = null;
+		// call under test
+		aclDAO.canAccess(gs, benefactors, ObjectType.ENTITY, ACCESS_TYPE.READ);
+	}
```
Comment (brucehoff) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/DBOAccessControlListDAOImplTest.java:244]: please add test(s) in which the passed in sets are empty

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java [hunk 3]
```diff
@@ -408,5 +414,22 @@ public AuthorizationStatus canCreateWiki(UserInfo userInfo, String objectId, Obj
...
+
+	@Override
+	public Set<Long> canReadBenefactors(UserInfo userInfo, Set<Long> benefactors) {
+		Set<Long> results = null;
+		if (userInfo.isAdmin()){
+			// admin same as input
+			results = Sets.newHashSet(benefactors);
+		}else{
+			// non-adim run a query
+			results = this.aclDAO.canAccess(userInfo.getGroups(), benefactors,
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java:427]: non-admin
canAccess -> getAccessibleBenefactors

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/NodeManagerImpl.java [hunk 4]
```diff
@@ -586,19 +600,32 @@ public EntityHeader getNodeHeader(UserInfo userInfo, String entityId, Long versi
...
+			// nothing to do.
+			return toFilter;
+		}
+		Set<Long> originalBenefactors = new HashSet<Long>(toFilter.size());
+		for(EntityHeader header: toFilter){
+			ValidateArgument.required(header.getBenefactorId(), "entityHeader.benefactorId");
+			originalBenefactors.add(header.getBenefactorId());
+		}
+		// find the intersection.
+		Set<Long> benefactorIntersection = authorizationManager.canReadBenefactors(userInfo, originalBenefactors);
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/NodeManagerImpl.java:621]: canReadBenefactors -> getAccessibleBenefactors
